source/kraken/tests/realtime_test.cpp

```diff
@@ -3340,3 +3341,167 @@ BOOST_FIXTURE_TEST_CASE(trip_id_that_doesnt_exist_must_be_rejected_in_classical_
+    BOOST_REQUIRE_EQUAL(pt_data.meta_vjs.exists("vj_new_trip_2"), true);
+
+    // Check vj table
+    BOOST_REQUIRE_EQUAL(pt_data.vehicle_journeys_map.size(), 3);
+    BOOST_REQUIRE_EQUAL(pt_data.vehicle_journeys.size(), 3);
+
+    transit_realtime::TripUpdate remove_new_trip = ntest::make_delay_message("vj_new_trip",
+        "20190101",
+        {
+            RTStopTime("stop_point:A", "20190101T0900"_pts),
+            RTStopTime("stop_point:E", "20190101T0930"_pts),
+            RTStopTime("stop_point:F", "20190101T1000"_pts),
+            RTStopTime("stop_point:G", "20190101T1030"_pts),
+        },
+        transit_realtime::Alert_Effect::Alert_Effect_NO_SERVICE,
+        comp_uri,
+        phy_mode_uri);
+
+    navitia::handle_realtime("feed-1", timestamp, remove_new_trip, *b.data, true, true);
+    b.data->build_raptor();
+
+    // trip from A to G is now removed
+    res = compute("20190101T073000", "stop_point:A", "stop_point:G");
+    BOOST_CHECK_EQUAL(res.response_type(), pbnavitia::NO_SOLUTION);
+    BOOST_CHECK_EQUAL(res.journeys_size(), 0);
+
+    // Check meta vj table
+    BOOST_REQUIRE_EQUAL(pt_data.meta_vjs.size(), 3);
+    BOOST_REQUIRE_EQUAL(pt_data.meta_vjs.exists("vj:1"), true);
+    BOOST_REQUIRE_EQUAL(pt_data.meta_vjs.exists("vj_new_trip"), true);
+    BOOST_REQUIRE_EQUAL(pt_data.meta_vjs.exists("vj_new_trip_2"), true);
+
+    // Check vj table
+    BOOST_REQUIRE_EQUAL(pt_data.vehicle_journeys_map.size(), 2);
+    BOOST_REQUIRE_EQUAL(pt_data.vehicle_journeys.size(), 2);
+
+    // Send the same GTFS-RT. Have to reject it because id doesn't exist.
+    remove_new_trip = ntest::make_delay_message("vj_new_trip",
+        "20190101",
+        {
+            RTStopTime("stop_point:A", "20190101T0900"_pts),
+            RTStopTime("stop_point:E", "20190101T0930"_pts),
+            RTStopTime("stop_point:F", "20190101T1000"_pts),
+            RTStopTime("stop_point:G", "20190101T1030"_pts),
+        },
+        transit_realtime::Alert_Effect::Alert_Effect_NO_SERVICE,
+        comp_uri,
+        phy_mode_uri);
+
+    navitia::handle_realtime("feed-1", timestamp, remove_new_trip, *b.data, true, true);
+    b.data->build_raptor();
+
+    // Check meta vj table
+    BOOST_REQUIRE_EQUAL(pt_data.meta_vjs.size(), 3);
+
+    // Check vj table
+    BOOST_REQUIRE_EQUAL(pt_data.vehicle_journeys_map.size(), 2);
+    BOOST_REQUIRE_EQUAL(pt_data.vehicle_journeys.size(), 2);
+}
+
+BOOST_FIXTURE_TEST_CASE(cant_cancel_trip_that_doesnt_exist, AddTripDataset) {
+
+    auto& pt_data = *b.data->pt_data;
+
+    BOOST_CHECK(!pt_data.meta_vjs.exists("vj_id_doesnt_exist"));
+    BOOST_CHECK_EQUAL(pt_data.meta_vjs.size(), 1);
+    BOOST_REQUIRE_EQUAL(pt_data.vehicle_journeys_map.size(), 1);
+    BOOST_REQUIRE_EQUAL(pt_data.vehicle_journeys.size(), 1);
+
+    // Can't cancel trip that doesn't exist
+    transit_realtime::TripUpdate remove_trip = ntest::make_delay_message("vj_id_doesnt_exist",
```

Comment:
note to self: rename this `make_delay_message` that does more than just delays now.